.github/workflows/CD.yml

```diff
@@ -629,16 +629,42 @@ jobs:
           src: public.ecr.aws/${{ env.ECR_REPO }}:${{ needs.release-checking.outputs.version }}
           dst: ${{ env.IMAGE_NAMESPACE }}/${{ env.IMAGE_NAME }}:${{ needs.release-checking.outputs.version }}
 
+  mirror-adot-operator:
+    runs-on: ubuntu-latest
+    needs: release-checking
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Cache if success
+        id: mirror-adot-operator
+        uses: actions/cache@v2
+        with:
+          key: mirror-adot-operator-${{ github.run_id }}
+          path: VERSION
+
+      - name: Set up Go 1.x
+        if: steps.mirror-adot-operator.outputs.cache-hit != 'true'
+        uses: actions/setup-go@v2
+        with:
+          go-version: ^1.17.7
+
+      - name: Compare version with Dockerhub latest
+        id: version
+        if: steps.mirror-adot-operator.outputs.cache-hit != 'true'
+        run: |
+          TAG="${{ needs.release-checking.outputs.version }}"
+          TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
+
       - name: Configure AWS Credentials
-        if: steps.release-latest-image.outputs.cache-hit != 'true'
+        if: steps.mirror-adot-operator.outputs.cache-hit != 'true'
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.RELEASE_KEY_ID }}
           aws-secret-access-key: ${{ secrets.RELEASE_SECRET }}
           aws-region: us-west-2
 
       - name: Mirror ADOT operator
-        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (steps.release-latest-image.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.same-version == 'true') }}
+        if: ${{ github.event.inputs.skip-adot-operator-mirroring == 'false' && (steps.mirror-adot-operator.outputs.cache-hit != 'true' && (steps.version.outputs.major-update == 'true' || steps.version.outputs.minor-update == 'true') || steps.version.outputs.patch-update == 'true') }}
```

Comment:
Why is the ADOT Operator release tied to the collector version release anyways? I think we could just get rid of the whole version check anyways. We should maybe see if anyone else has input on this. 

Is there a situation, other than when it is explicitly disabled, that we do not want to mirror the ADOT operator?

Reply:
I'm okay with getting rid of the version check; the tool, when run, will not re-mirror existing Operator images regardless, but yeah some outside input might be helpful. The ADOT Operator release is tied to the Collector release because we would want to use the same version of the Operator/Collector (i.e 0.17 ADOT Collector (0.45 OTEL Collector) + 0.45 ADOT Operator)

There isn't really a situation where we would NOT want to mirror the ADOT Operator, especially when considering that it doesn't re-mirror images.